docs/docs/reference/changed/vararg-patterns.md

```diff
@@ -0,0 +1,21 @@
+---
+layout: doc-page
+title: "Vararg Patterns"
+---
+
+The syntax of vararg patterns has changed. In the new syntax one
+writes varargs in patterns exactly like one writes them in
+expressions, using a `: _*` type annotation:
+
+    xs match {
+      case List(1, 2, xs: _*) => println(xs)    // binds xs
+      case List(1, _ : _*) =>                   // wildcard pattern
+    }
+
+The old syntax, which is shorter but less regular, is no longer
+supported:
+
+    /*!*/ case List(1, 2, xs @ _*)       // syntax error
+    /*!*/ case List(1, 2, _*) => ...     // syntax error
+
```

Comment:
@xeno-by AFAIK, we support both syntaxes under -language:Scala2
https://github.com/lampepfl/dotty/issues/1059

On 30 May 2017 01:14:25 Eugene Burmako <notifications@github.com> wrote:

> xeno-by commented on this pull request.
>
>
>
>> +
> +The syntax of vararg patterns has changed. In the new syntax one
> +writes varargs in patterns exactly like one writes them in
> +expressions, using a `: _*` type annotation:
> +
> +    xs match {
> +      case List(1, 2, xs: _*) => println(xs)    // binds xs
> +      case List(1, _ : _*) =>                   // wildcard pattern
> +    }
> +
> +The old syntax, which is shorter but less regular, is no longer
> +supported:
> +
> +    /*!*/ case List(1, 2, xs @ _*)       // syntax error
> +    /*!*/ case List(1, 2, _*) => ...     // syntax error
> +
>
> When developing Scalameta 2.x that's supposed to crosscompile against 2.x 
> and 3.x, I found this to be an unfortunate problem. There's currently no 
> way to write crosscompilable code that contains sequence wildcards. Can 
> something be done with this?
>
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/lampepfl/dotty/pull/2592#pullrequestreview-40833029